analyses/sample-distribution-analysis/03-tumor-descriptor-and-assay-count.Rmd

```diff
@@ -0,0 +1,263 @@
+
+How many participants have paired genomic and transcriptomic samples for the same `tumor_descriptor` values?
+
+```{r}
+# if we perform a full join here using the pariticpant ID and descriptors, we 
+# will get NAs in columns where pairs don't exist and can use this information 
+# to count
+paired_df <-
+  full_join(genomic_df, transcriptomic_df,
+                   by = c("Kids_First_Participant_ID", "descriptors"))
+
+head(paired_df, 10) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+#### All paired
+
+Count the examples where all time points have both kinds of assays.
+
+```{r}
+# no NAs = both kinds of assays are present
+paired_df %>%
+  filter(complete.cases(.)) %>%
+  group_by(descriptors) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part="all")
+```
+
+#### No RNA-seq
+
+If the `experimental_strategy.y` column has an `NA`, RNA-seq is missing for that participant-descriptors pair.
+
+```{r}
+paired_df %>%
+  filter(is.na(experimental_strategy.y)) %>%
+  group_by(descriptors) %>%
+  tally() %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+#### No WXS or WGS
+
+If the `experimental_strategy.x` column has an `NA`, there is no WGS or WXS for that participant-descriptors pair.
+
+```{r}
+paired_df %>%
+  filter(is.na(experimental_strategy.x)) %>%
+  group_by(descriptors) %>%
+  tally() %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+#### Duplicate participant IDs
+
+There are cases where one timepoint (e.g., `tumor_descriptor` value) is missing from one kind of assay but not the other.
+This will show up as duplicated values in the `Kids_First_Participant_ID` column.
+
+```{r}
+ids_with_dups <- paired_df %>%
+  filter(duplicated(Kids_First_Participant_ID)) %>%
+  pull(Kids_First_Participant_ID)
+```
+
+There are `r length(ids_with_dups)` cases of this.
+Let's see what this looks like.
+
+```{r}
+paired_df %>%
+  filter(Kids_First_Participant_ID %in% ids_with_dups) %>%
+  arrange(Kids_First_Participant_ID)  %>% 
+  regulartable() %>%
+  fontsize(size = 12, part = "all")
+```
+
+*Two examples for interpretation:*
+
+So for the `PT_3KM9W8S8` participant ID, there is primary WGS and RNA-seq data, but only RNA-seq data for progressive disease.
+For `PT_HT4HJXY6`, there is WGS data for both the primary CNS tumor and the second malignancy, but RNA-seq data only for the second malignancy.
+
+## By histology
+
+We're going to use the `disease_type_new` column here.
+
+### Primary only
+
+We're including *any* assay type.
+This table is different than what is plotted upstream in this module because we didn't restrict that to `Initial CNS Tumor` only.
+
+```{r}
+tumor_df %>%
+  filter(tumor_descriptor == "Initial CNS Tumor") %>%
+  distinct(Kids_First_Participant_ID, disease_type_new) %>%
+  group_by(disease_type_new) %>%
+  tally() %>%
+  arrange(desc(n)) %>% 
+  regulartable() %>%
+  fontsize(size = 12, part="all")
+```
+
```

Comment:
I have questions about some of these descriptors, and the meaning of delimiters. For example, we have: `Ganglioglioma;Low-grade glioma/astrocytoma (WHO grade I/II)`, as well as `Ganglioglioma` on its own, and `Low-grade glioma;astrocytoma (WHO grade I/II)` on its own. Is the first some kind of combination of the latter two? Is there a semantic difference between the use of `;` and `/` in these cases?  This PR is probably not the right place for this discussion...

Reply:
Hmm, these are all derived from the way they were added to the database - seems the `\` should not be between LGG/astrocytoma, but that should be a `;`.  Ganglioglioma and LGG/astrocytoma are different and only shown together since they were both on the pathology report. If possible, through genomic analyses, we could further dial into the diagnoses, we can possibly replace with one or the other.